tests/tests/test-staking.ts

```diff
@@ -791,6 +792,176 @@ describeDevMoonbeam("Staking - Rewards", (context) => {
   });
 });
 
+describeDevMoonbeam("Staking - Delegator Leave", (context) => {
+  const EXTRA_BOND_AMOUNT = 1_000_000_000_000_000_000n;
+  const BOND_AMOUNT = MIN_GLMR_NOMINATOR + EXTRA_BOND_AMOUNT;
+
+  let ethan: KeyringPair;
+  let balathar: KeyringPair;
+
+  before("should set baltathar as candidate", async () => {
+    const keyring = new Keyring({ type: "ethereum" });
+    ethan = keyring.addFromUri(ETHAN_PRIVKEY, null, "ethereum");
+    balathar = keyring.addFromUri(BALTATHAR_PRIV_KEY, null, "ethereum");
+    await context.polkadotApi.tx.parachainStaking
+      .joinCandidates(MIN_GLMR_STAKING, 1)
+      .signAndSend(balathar);
+    await context.createBlock();
+  });
+
+  beforeEach(
+    "should successfully call delegate on ALITH and BALTATHAR and schedule leave",
+    async () => {
+      await context.polkadotApi.tx.parachainStaking
+        .delegate(ALITH, BOND_AMOUNT, 0, 0)
+        .signAndSend(ethan);
+      await context.createBlock();
+
+      await context.polkadotApi.tx.parachainStaking
+        .delegate(BALTATHAR, BOND_AMOUNT, 0, 1)
+        .signAndSend(ethan);
+      await context.createBlock();
+
+      await context.polkadotApi.tx.parachainStaking.scheduleLeaveDelegators().signAndSend(ethan);
+      await context.createBlock();
+      const currentRound = await context.polkadotApi.query.parachainStaking.round();
+      await jumpToRound(context, currentRound.current.addn(1).toNumber());
+    }
+  );
+
+  afterEach("should clean up delegation requests", async () => {
+    await context.polkadotApi.tx.parachainStaking.cancelLeaveDelegators().signAndSend(ethan);
+    await context.createBlock();
+    await context.polkadotApi.tx.parachainStaking.cancelDelegationRequest(ALITH).signAndSend(ethan);
+    await context.createBlock();
+    await context.polkadotApi.tx.parachainStaking
+      .cancelDelegationRequest(BALTATHAR)
+      .signAndSend(ethan);
+    await context.createBlock();
+  });
+
+  it("should allow executeLeaveDelegators to be executed after round delay", async function () {
+    this.timeout(20000);
```

Comment:
Why such a large timeout? Is that necessary when using `jumpToRound`?

Reply:
Exactly. The nature of testing this functionality calls for jumping 2 or more rounds, which have a length of some 100/600 or so blocks iirc. Unfortunate.

Reply:
You can change the round length to some smaller value before calling this. There's a compile time constant minimum of 10 for moonbase though.